multichain-testing/test/xcs-swap-anything/helpers.ts

```diff
@@ -0,0 +1,826 @@
+        const { label } = xcsContracts[contract];
+        wasmFiles.push(`${label}.wasm`);
+      }
+    }
+
+    try {
+      await execa('bash', [
+        scriptPath,
+        'osmosis-labs',
+        'osmosis',
+        branch,
+        'tests/ibc-hooks/bytecode',
+        artifactsPath,
+        ...wasmFiles,
+      ]);
+    } catch (error) {
+      throw Error(`Failed to download XCS artifacts: ${error}`);
+    }
+  };
+
+  const instantiateSingleXcsContract = async (
+    contractLabel: string,
+    instantiateArgs: CommonArgs | CrosschainSwapsArgs,
+  ) => {
+    const clientRegistry = osmosisClient.registry;
+    clientRegistry.register(MsgStoreCode.typeUrl, MsgStoreCode);
+
+    clientRegistry.register(
+      MsgInstantiateContract.typeUrl,
+      MsgInstantiateContract,
+    );
+
+    const contractPath = `${xcsArtifactsPath}/${contractLabel}.wasm`;
+
+    const wasmPath = path.resolve(contractPath);
+    const wasmByteCode = await fs.readFile(wasmPath);
+
+    const fee = {
+      amount: [{ denom: 'uosmo', amount: '100000' }],
+      gas: '10000000',
+    };
+
+    const storeMessage = MsgStoreCode.fromPartial({
+      sender: osmosisAddress,
+      wasmByteCode,
+    });
+
+    const storeEncodeObjects: Array<
+      import('@cosmjs/proto-signing').EncodeObject
+    > = [
+      {
+        typeUrl: MsgStoreCode.typeUrl,
+        value: storeMessage,
+      },
+    ];
+
+    const storeResult = await osmosisClient.signAndBroadcast(
+      osmosisAddress,
+      storeEncodeObjects,
+      fee,
+    );
+
+    if (storeResult.msgResponses[0] && storeResult.code !== 0) {
+      throw Error(`Failed to store ${contractLabel} contract`);
+    }
+
+    const codeId = MsgStoreCodeResponse.decode(
+      storeResult.msgResponses[0].value,
+    ).codeId;
+
+    const instantiateMessage = MsgInstantiateContract.fromPartial({
+      sender: osmosisAddress,
+      admin: osmosisAddress,
+      codeId,
+      label: contractLabel,
+      msg: toUtf8(JSON.stringify(instantiateArgs)),
+    });
+
+    const instantiateEncodeObjects: Array<
+      import('@cosmjs/proto-signing').EncodeObject
+    > = [
+      {
+        typeUrl: MsgInstantiateContract.typeUrl,
+        value: instantiateMessage,
+      },
+    ];
+
+    const instantiateResult = await osmosisClient.signAndBroadcast(
+      osmosisAddress,
+      instantiateEncodeObjects,
+      fee,
+    );
+
+    if (instantiateResult.msgResponses[0] && instantiateResult.code !== 0) {
+      throw Error(`Failed to instantiate ${contractLabel} contract`);
+    }
+
+    const address = MsgInstantiateContractResponse.decode(
+      instantiateResult.msgResponses[0].value,
+    ).address;
+
+    return { codeId, address };
+  };
+
+  const persistXcsInfo = async () => {
+    try {
+      const sanitizedContracts = JSON.parse(
+        JSON.stringify(xcsContracts, (_, v) =>
+          typeof v === 'bigint' ? Number(v) : v,
+        ),
```

Comment:
the bigint replacer is used twice. I suggest factoring it out:

```suggestion
        JSON.stringify(xcsContracts, bigintReplacer),
```

where

```ts
const bigintReplacer = (_, v) =>  typeof v === 'bigint' ? Number(v) : v;
```

Reply:
Change request implemented on commit: 9bdf4e3ca6d289d459fa0b7d514776e6dfe20231